### ISSUE TITLE
MPI_File_open not thread safe

### ISSUE THREAD

**Issue by nysal** [open-mpi/ompi#1762]
MPI_File_open() loads the io framework lazily, which is not thread safe

``` C
    rc = mca_base_framework_open(&ompi_io_base_framework, 0);
    if (OMPI_SUCCESS != rc) {
        return OMPI_ERRHANDLER_INVOKE(MPI_FILE_NULL, rc, FUNC_NAME);
    }
```

We should probably protect this with a mutex. Should we just use the bootstrap mutex used in MPI_Init ? The same problem exists in a few other areas too - MPI_File_delete, MPI_Register_datarep

**Comment by hppritcha** [comment]
@edgargabriel should this be assigned to you?  If so, which milestone would you suggest?

**Comment by hppritcha** [comment]
@edgargabriel says this will not be fixed in the 2.0x/2.x release branches.  It is fixed in 3.0.x.
